### ISSUE TITLE
Infinite Loop in the Transaction Profiler 

### ISSUE THREAD

**Issue by mpass99** [getsentry/sentry-go#724]
## Summary

The transaction profiler `profiler.go` loops infinite if the transaction duration exceeds the `profilerRuntimeLimit`, causing the application to terminate due to OOM.

## Steps To Reproduce

```go
func main() {
	err := sentry.Init(sentry.ClientOptions{EnableTracing: true, ProfilesSampleRate: 1})
	if err != nil {
		fmt.Printf("sentry.Init: %s", err.Error())
		return
	}

	span := sentry.StartSpan(context.Background(), "main", sentry.WithSpanSampled(sentry.SampledTrue))
	time.Sleep(35 * time.Second)
	span.Finish()
}
```

## Expected Behavior

Either:
- The Sentry SDK is able to profile also transactions lasting longer than 30 seconds without crashing the application.
- The Sentry SDK discards the profiling of transactions exceeding the `profilerRuntimeLimit`.

### SDK
 - `sentry-go` version: v0.24.1
 - Go version: 1.21.1
 - Using Go Modules? yes

### Sentry
 - No Sentry Backend required to reproduce the behavior.

## Additional context

The following loop runs infinite:
https://github.com/getsentry/sentry-go/blob/7e37edeb9589c5e8966c5c80822aa4ddaa0fbfd2/profiler.go#L243-L251

The `profilerRuntimeLimit` defines a limit of 30 seconds. If the transaction lasts longer, the `start`-30-second-ring-structure is filled completely which causes both break conditions to not apply.

**Comment by cleptric** [comment]
@vaind Can you take a look, please?

**Comment by cjcormack** [comment]
#725 fixes this issue for me. Thank you for getting a PR with the fix out so quickly!

**Comment by cleptric** [comment]
The fix was released in [0.25.0](https://github.com/getsentry/sentry-go/releases/tag/v0.25.0). Thanks for confirming the fix @cjcormack 🙏

**Comment by mpass99** [comment]
I can confirm that release `0.25.0` resolves our issue. Thank you for your quick solution!
